### ISSUE TITLE
libtorch model load and inference slow when running first time

### ISSUE THREAD

**Issue by lsd1994** [pytorch/pytorch#44269]
When I load and inference a model, it runs slow at first time, but  returns to normal speed after first time. 
Is it normal for libtorch or I miss something basic?
I use win10+libtorch1.5+cuda10.1+vs2017

code:
void f()
{
	clock_t t1, t2, t3;
	t1 = clock();
	torch::jit::script::Module model = torch::jit::load("E:/python/pytorch/model.pt");; 
	model.eval();
	model.to(torch::kCUDA);
	t2 = clock();
	printf("load time : %f ms\n", (double)(t2 - t1) * 1000 / CLOCKS_PER_SEC);
	vector<torch::jit::IValue> inputs;
	torch::Tensor t = torch::ones({ 1,3,127,127 });
	inputs.push_back(t.to(torch::kCUDA));
	auto out_tensor = model.forward(inputs).toTensor().cpu();
	t3 = clock();
	printf("inference time : %f ms\n\n", (double)(t3 - t2) * 1000 / CLOCKS_PER_SEC);
}
int main()
{
	f(); f(); f(); f(); f();
	return 0;
}

result:
![1](https://user-images.githubusercontent.com/29648659/92343475-6e37e780-f0f6-11ea-83ea-30a2ff4d22fa.jpg)

**Comment by kshitij12345** [comment]
I think it has to do with initialising CUDA related stuff. 

To mitigate, you can mostly push a dummy input once so that you can have inference at the usual speed.

**Comment by lsd1994** [comment]
@kshitij12345 Thank for your reply. I will try this.

**Comment by robieta** [comment]
> I think it has to do with initialising CUDA related stuff.
> 
> To mitigate, you can mostly push a dummy input once so that you can have inference at the usual speed.

Indeed, lazy CUDA init is expected. @kshitij12345's suggestion of a warmup is the recommended solution.

**Comment by WANGCHAO1996** [comment]
> Indeed, lazy CUDA init is expected. @kshitij12345's suggestion of a warmup is the recommended solution.
Hello, my cuda10.2 win10 libtorch1.7.1 the first and second reasoning speed is very slow, the second is the slowest, then normal. What's the reason, thank you
![图片](https://user-images.githubusercontent.com/51087601/105497969-89e50880-5cfa-11eb-8000-a5ae285189b2.png)

**Comment by Xavier31** [comment]
I have the same behavior as @WANGCHAO1996 : the second iteration is much slower. I understand the warm-up for the first one, but the second seems really weird. I have this behavior with different models, on linux/Windows and on different libtorch version (tested 1.10 and 1.5)

CUDA available! running on GPU
Processing time: 239 ms
Processing time: 6591 ms
Processing time: 29 ms
Processing time: 29 ms
Processing time: 28 ms
Processing time: 29 ms

**Comment by Eugene1518** [comment]
> I think it has to do with initialising CUDA related stuff.
> 
> To mitigate, you can mostly push a dummy input once so that you can have inference at the usual speed.

very good
